src/Http/Client.php

```diff
@@ -34,37 +40,53 @@ public function __construct(RequesterInterface $requester, ?iterable $requestDec
     }
 
     /** {@inheritdoc} */
-    public function delete(string $path, ?array $headers = null): Response
+    public function delete(string $path, ?array $headers = null): ResponseInterface
     {
         return $this->request('DELETE', $path, null, $headers);
     }
 
     /** {@inheritdoc} */
-    public function get(string $path, ?array $headers = null): Response
+    public function get(string $path, ?array $headers = null): ResponseInterface
     {
         return $this->request('GET', $path, null, $headers);
     }
 
+    /**
+     * Performs a merge-patch request to the API service using the PATCH method.
+     *
+     * @param array|string|resource|Closure<string>|Traversable<string>|null $requestData
+     * @param string[]|string[][]|null                                       $headers
+     *
+     * @phpstan-param array|string|resource|Closure(): string|Traversable<string>|null $requestData
+     */
+    public function mergePatch(string $path, $requestData = null, ?array $headers = null): ResponseInterface
```

Comment:
I think that a merge-patch operation should be generated in a more structured way (or at least a separate pull request).
Let's open a new issue and continue there.

Reply:
Seems legit :) removed from the Client, opened a more specific issue #2